packages/debug/src/browser/debug-frontend-application-contribution.ts

```diff
@@ -611,11 +611,17 @@ export class DebugFrontendApplicationContribution extends AbstractViewContributi
             execute: (config?: DebugSessionOptions) => this.start(true, config)
         });
         registry.registerCommand(DebugCommands.STOP, {
-            execute: () => this.manager.currentSession && this.manager.currentSession.terminate(),
+            execute: () => {
+                this.manager.updateCurrentSession(undefined);
+                return this.manager.currentSession && this.manager.currentSession.terminate();
```

Comment:
Should it not be the responsibility of the manager to update the current session, rather than callers?
I do not know if outside callers should be able to simply clear the session.

Reply:
changed it so manager will update current session.